### ISSUE TITLE
lib/js-asset_paths.rb symlink pointing to wrong place in packaged .gem (v0.3.0)

### ISSUE THREAD

**Issue by sbleon** [sonnym/js-asset_paths#7]
There are two`require`able entry points in the `lib` dir:

- `js-asset_paths.rb`
- `js_asset_paths.rb`

As far as I can tell, the former has always been a symlink to the latter. It seems that something changed in the packaging between v0.2.0 and v0.3.0, resulting in a broken symlink in the packaged version of the gem.

In v0.2.0, there is no symlink:
```
$ ls -al ~/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/js-asset_paths-0.2.0/lib
total 16
drwxr-xr-x   5 leon  staff  170 Jun  8 15:27 .
drwxr-xr-x  11 leon  staff  374 Jun  8 15:27 ..
-rw-r--r--   1 leon  staff  100 Jun  8 15:27 js-asset_paths.rb
drwxr-xr-x   5 leon  staff  170 Jun  8 15:27 js_asset_paths
-rw-r--r--   1 leon  staff  100 Jun  8 15:27 js_asset_paths.rb
```

In v0.3.0, there is a symlink, but it points to a non-existent file:
```
$ ls -al ~/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/js-asset_paths-0.3.0/lib
total 16
drwxr-xr-x   5 leon  staff  170 Jun  8 12:57 .
drwxr-xr-x  12 leon  staff  408 Jun  8 12:57 ..
lrwxr-xr-x   1 leon  staff   21 Jun  8 12:57 js-asset_paths.rb -> lib/js_asset_paths.rb
drwxr-xr-x   5 leon  staff  170 Jun  8 12:57 js_asset_paths
-rw-r--r--   1 leon  staff  100 Jun  8 12:57 js_asset_paths.rb
```

Oddly, the symlink in the repo appears correct:
```
$ ls -al ~/code/gems/js-asset_paths/lib
total 16
drwxr-xr-x   5 leon  staff  170 Aug 11  2016 .
drwxr-xr-x  19 leon  staff  646 Jun  8 12:31 ..
lrwxr-xr-x   1 leon  staff   17 Aug 11  2016 js-asset_paths.rb -> js_asset_paths.rb
drwxr-xr-x   5 leon  staff  170 Jun  8 12:31 js_asset_paths
-rw-r--r--   1 leon  staff  100 Aug 11  2016 js_asset_paths.rb
```

It appears as if something about the `gem build` command may be breaking the symlink. The resulting error (when compiling a `.js` file that `require`s `js-asset_paths`) is `Sprockets::FileNotFound: couldn't find file 'js-asset_paths' with type 'application/javascript'`.

**Workaround:**
`gem 'js-asset_paths', require: 'js_asset_paths'`

**Comment by sonnym** [comment]
Curious - I wonder when that was introduced.  When building the gem, I was getting the following:  `WARNING:  lib/js-asset_paths.rb is a symlink, which is not supported on all platforms`.  Unfortunately, the gem specification does not allow a custom entry point, but it looks like it handles hard links well enough.  Can you try the following branch and let me know if that fixes the issue?  Thanks!  https://github.com/sonnym/js-asset_paths/tree/symlink_fix

**Comment by sbleon** [comment]
Thanks for looking into this!

It looks like what was a hard link on your local filesystem ended up being
just a copy of the file in git. I don't think git understand hard links. In
any case, that branch eliminates the need to specify an entry point, which
was my goal.

My recommendation would be to leave `js-asset_paths.rb` as it is now, since
that file will be `require`d by default, and have `js_asset_paths.rb` (the
alternate entry point) just contain `require_relative 'js-asset_paths'` to
preserve backwards compatibility. We could even put a deprecation warning
in the alternate entry point.

**Comment by sonnym** [comment]
That makes sense - I went with that approach, minus deprecating the alternate entry point, since it should neither be a problem nor incur any maintenance overhead.  These changes are released in 0.3.1.  I also cut a 0.4.0 with some other changes, most notably raising a `TypeError` when `config.assets.digest` is not a boolean.
